kubernetes/ray-cluster.yaml

```diff
@@ -0,0 +1,127 @@
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: ray
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: ray-head
+  namespace: ray
+spec:
+  ports:
+    - name: redis-primary
+      port: 6379
+      targetPort: 6379
+    - name: redis-shard-0
+      port: 6380
+      targetPort: 6380
+    - name: redis-shard-1
+      port: 6381
+      targetPort: 6381
+    - name: object-manager
+      port: 12345
+      targetPort: 12345
+    - name: node-manager
+      port: 12346
+      targetPort: 12346
+  selector:
+    component: ray-head
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: ray-head
+  namespace: ray
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      component: ray-head
+      type: ray
+  template:
+    metadata:
+      labels:
+        component: ray-head
+        type: ray
+    spec:
+      volumes:
+      - name: dshm
+        emptyDir:
+          medium: Memory
+      containers:
+        - name: ray-head
+          image: rayproject/autoscaler
+          command: [ "/bin/bash", "-c", "--" ]
+          args: 
+            - "ray start --head --node-ip-address=$MY_POD_IP --redis-port=6379 --redis-shard-ports=6380,6381 --num-cpus=$MY_CPU_REQUEST --object-manager-port=12345 --node-manager-port=12346 --block"
+          ports:
+            - containerPort: 6379
+            - containerPort: 6380
+            - containerPort: 6381
+            - containerPort: 12345
+            - containerPort: 12346
+          volumeMounts:
+            - mountPath: /dev/shm
+              name: dshm
+          env:
+            - name: MY_POD_IP
+              valueFrom:
+                fieldRef:
+                  fieldPath: status.podIP
+            - name: MY_CPU_REQUEST
+              valueFrom:
+                resourceFieldRef:
+                  resource: requests.cpu
+          resources:
+            limits:
+              cpu: 1000m
+              memory: 1Gi
+---
+apiVersion: apps/v1
+kind: Deployment
```

Comment:
I think we should use pod instead of deployment here. Because the deployment controller will create new pod if the autoscaler deleted it.